source/renderer/dialog/note-network.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        NoteNetwork class
+ * CVM-Role:        View
+ * Maintainer:      Julien Mirval
+ * License:         GNU GPL v3
+ *
+ * Description:     This dialog lets you visualize all notes as a network.
+ *
+ * END HEADER
+ */
+
+const d3 = require('d3')
+
+const ZettlrDialog = require('./zettlr-dialog.js')
+
+class NoteNetwork extends ZettlrDialog {
+  constructor () {
+    super()
+    this._dialog = 'note-network'
+    this.graph = []
+  }
+
+  preInit (data) {
+    var tmpLinks = []
+    var tmpNodes = []
+
+    // Building link list
+    for (const a of Object.entries(data)) {
+      if (a[1].outbound) {
+        for (const i of a[1].outbound) {
+          if (i) {
+            tmpLinks.push({ 'source': a[0], 'target': i })
+          }
+        }
+      }
+    }
+
+    // Building node list
+    for (const a of Object.entries(data)) {
+      tmpNodes.push({
+        'id': a[0],
+        'title': a[1].name ? a[1].name.replace(a[0], '') : 'Undefined',
+        'inbound': a[1].inbound ? a[1].inbound.length : 0,
+        'outbound': a[1].outbound ? a[1].outbound.length : 0
+      })
+    }
+
+    this.graph = {
+      'nodes': tmpNodes,
+      'links': tmpLinks
+    }
```

Comment:
In the "Add nodes" part, you're calling `.attr('r', function (d) { return 1 + d.inbound + d.outbound })`.
When I first read it, I was like, "Aren't `inbound` and `outbound` arrays? Wouldn't calling this code return incorrect results like the following snippet?"
```js
(1 + [2, 3] + [4, 5]) === '12,34,5'
```

But it seems like `LinkProvider's Node/Link` objects are different from NoteNetwork's. In this case, I suggest using different names for them... by "them," I mean the types/objects, or at least rename the attributes from `inbound` to `inboundCount` and `outbound` to `outboundCount`.